File: scratch/dedup.py
import yaml
import sys

def deduplicate_yaml(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        data = yaml.safe_load(f)
    
    tracked = data.get('tracked_companies', [])
    seen_names = set()
    unique_tracked = []
    
    for company in tracked:
        name = company.get('name')
        if name not in seen_names:
            unique_tracked.append(company)
            seen_names.add(name)
        else:
            print(f"Removing duplicate: {name}")
            
    data['tracked_companies'] = unique_tracked
    
    with open(file_path, 'w', encoding='utf-8') as f:
        yaml.dump(data, f, sort_keys=False, allow_unicode=True)

if __name__ == "__main__":
    deduplicate_yaml(sys.argv[1])

File: scratch/dedup.mjs
import fs from 'fs';
import yaml from 'js-yaml';
import path from 'path';

const filePath = process.argv[2];
const content = fs.readFileSync(filePath, 'utf8');
const data = yaml.load(content);

const tracked = data.tracked_companies;
const seen = new Set();
const unique = [];

for (const company of tracked) {
  if (!seen.has(company.name)) {
    unique.push(company);
    seen.add(company.name);
  } else {
    console.log(`Removing duplicate: ${company.name}`);
  }
}

data.tracked_companies = unique;
fs.writeFileSync(filePath, yaml.dump(data, { noRefs: true, lineWidth: -1 }));
console.log('Deduplication complete.');
